src/item.h

```diff
@@ -85,6 +88,96 @@ class item_category
+        // items stored in this bag
+        std::vector<item> items;
+        // holds any extra information we need about storage
+        std::map<std::string, void *> var;
+
+    public:
+        storage() {};
+        storage(item *self) : me(self) {}
+        storage(item *self, std::vector<item> &&item_vec)
+            : me(self), items(std::move(item_vec)) {}
+        storage(item *self, item_iterator start, item_iterator stop);
+
+        void init(item *self)                               {        me = self;         }
+        /*-----------------------------------------------------------------------------
+         *                                  OVERLOADS
+         *-----------------------------------------------------------------------------*/
+        item &operator[](size_t index)                      { return items[index];      }
+        const item &operator[](size_t index) const          { return items[index];      }
+        item_iterator begin()                               { return items.begin();     }
+        item_const_iterator begin() const                   { return items.cbegin();    }
+        item_iterator end()                                 { return items.end();       }
+        item_const_iterator end() const                     { return items.cend();      }
+        size_t size() const                                 { return items.size();      }
+        bool empty() const                                  { return items.empty();     }
+        void clear()                                        {        items.clear();     }
+        /*-----------------------------------------------------------------------------
+         *                                  ACCESS
+         *-----------------------------------------------------------------------------*/
+        // retrieve a reference to the items vector
+        std::vector<item> &get()                            { return items;             }
+        const std::vector<item> &get() const                { return items;             }
+        /*-----------------------------------------------------------------------------
+         *                                MANIPULATION
+         *-----------------------------------------------------------------------------*/
+        // return a pointer to where the item is now stored
+        // (use the new pointer to operate on object)
+        item *add(const item &thing);
+        // insert a set of items
+        std::vector<item *> add(std::vector<item> items, size_t index=0);
+        // insert a range of items
+        std::vector<item *> add(item_iterator start, item_iterator stop, size_t index=0);
+        // removes the item from storage, returning the item itself
+        item rem(size_t index);
+        // removes the item (from iterator)
+        item_iterator rem(item_iterator iter);
+        // removes a range of items
+        std::vector<item> rem(item_iterator start, item_iterator stop);
```

Comment:
Why not use the `std::vector` function names `push_back` and `erase`, you don't have to update all the code that uses it.